### ISSUE TITLE
`no-warning-comments` cannot be set in a configuration comment

### ISSUE THREAD

**Issue by IanVS** [eslint/eslint#3619]
If you try to set `no-warning-comments` with `"location": "anywhere"` in a configuration comment, it itself is flagged as an error:

```js
/*eslint no-warning-comments: [2, { "terms": ["todo", "fixme", "any other term"], "location": "anywhere" }]*/

var x = 10;
```

Results:
```
1:1  error  Unexpected todo comment            no-warning-comments
1:1  error  Unexpected fixme comment           no-warning-comments
1:1  error  Unexpected any other term comment  no-warning-comments
```

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by BYK** [comment]
:laughing:

**Comment by lo1tuma** [comment]
Any ideas how to fix this issue? Just ignoring all eslint comments in the rule would be a breaking change for me, because I use to rule to forbid eslint comments.

**Comment by BYK** [comment]
@lo1tuma uh oh, I didn't think that in my PR. That said how can you forbid eslint directives with this? You use `eslint` as the warning term?

**Comment by lo1tuma** [comment]
> You use eslint as the warning term?

Exactly. I know it doesn’t work for all eslint comments (e.g. `/* eslint-disable */` or `/* eslint-disable no-warning-comments */` will disable the rule so those comments can’t be detected anymore, see #1298), but it is still better than nothing.

**Comment by BYK** [comment]
@lo1tuma I see two options:

1. Convince @nzakas to add an option/rule to disable ESLint overrides
2. Add an option to this rule saying `includeDirectives`

What do you think?

**Comment by lo1tuma** [comment]
1. I also thought about a CLI option, but actually I like the ability to make this only a warning instead of an error.
2. This option would work for me. But I think it would be still a breaking change because you have to change the rule configuration to make it work the same way as before.

**Comment by BYK** [comment]
I'm unsure about this being a breaking change since it really is a bug if you ask me. I'll let others decide tho. We can easily make that option `true` by default to keep this backwards compatible.

**Comment by nzakas** [comment]
Oops, should be "or", not " out"

**Comment by lo1tuma** [comment]
I use it in a config file.

**Comment by nzakas** [comment]
Okay, so in that case, it seems like we should be able to make a special case where the rule says "if I see a comment that's configuring me, don't warn." Would that work?

Also, @lo1tuma, I think your use case is probably not correct for this rule. I'd suggest looking at creating a `no-eslint-comments` rule for that use case. Our configuration comments are special cases, and I think we need to treat them as such across the board.

**Comment by IanVS** [comment]
Perhaps @lo1tuma's use case calls for a separate rule `no-inline-config`?  Not sure if it would belong in core or a plugin.

**Comment by lo1tuma** [comment]
@nzakas So you are proposing that `no-warning-comments` should ignore `/* eslint [...]no-warning-coments[...] */` but not e.g. `/* eslint on-var: 2 */`? That would work for me.

I agree that it would make sense as a separate rule (it was already rejected to be a core rule, see #1298). But I think that the discussion of a separate rule for this use-case shouldn’t influence the discussion how we want to solve the current issue: with a breaking change, or without?

**Comment by nzakas** [comment]
@lo1tuma yup, that's my proposal.

Ah yeah, I remember that issue, comments are hard. Maybe what we really need is a command line flag that disables the use of configuration comments?
